tests/rptest/tests/redpanda_oauth_test.py

```diff
@@ -0,0 +1,155 @@
+        except Exception as e:
+            self.logger.error(f"{e}")
+            self.keycloak.clean_node(kc_node)
+            assert False, "Keycloak failed to start"
+
+        self.keycloak.admin.create_user('norma',
+                                        'desmond',
+                                        realm_admin=True,
+                                        email='10086@sunset.blvd')
+        self.keycloak.login_admin_user(kc_node, 'norma', 'desmond')
+        self.keycloak.admin.create_client(CLIENT_ID)
+
+        # add an email address to myapp client's service user. this should
+        # appear alongside the access token.
+        self.keycloak.admin.update_user(f'service-account-{CLIENT_ID}',
+                                        email='myapp@customer.com')
+
+        self.rpk.create_topic(EXAMPLE_TOPIC)
+        service_user_id = self.keycloak.admin_ll.get_user_id(
+            f'service-account-{CLIENT_ID}')
+        result = self.rpk.sasl_allow_principal(f'User:{service_user_id}',
+                                               ['all'], 'topic', EXAMPLE_TOPIC,
+                                               self.su_username,
+                                               self.su_password,
+                                               self.su_algorithm)
+
+        cfg = self.keycloak.generate_oauth_config(kc_node, CLIENT_ID)
+        assert cfg.client_secret is not None
+        assert cfg.token_endpoint is not None
+        k_client = PythonLibrdkafka(self.redpanda,
+                                    algorithm='OAUTHBEARER',
+                                    oauth_config=cfg)
+        producer = k_client.get_producer()
+
+        producer.produce(topic=EXAMPLE_TOPIC,
+                         key='bar',
+                         value='23',
+                         on_delivery=self.delivery_report)
+        producer.produce(topic=EXAMPLE_TOPIC,
+                         key='baz',
+                         value='23',
+                         on_delivery=self.delivery_report)
+        producer.produce(topic=EXAMPLE_TOPIC,
+                         key='qux',
+                         value='23',
+                         on_delivery=self.delivery_report)
+
+        # Expclicit poll triggers OIDC token flow. Required for librdkafka
+        # metadata requests to behave nicely.
+        producer.poll(0.0)
+
+        with expect_exception(KafkaException, lambda _: True):
+            producer.list_topics(timeout=5)
+
+        self.logger.info('Flushing {} records...'.format(len(producer)))
+
+        # TODO: Uncomment. Need to call flush or poll to force producer into OAUTH handshake
+        # producer.flush(0.0)
+
+        self.logger.debug(f"{self.produce_messages} {self.produce_errors}")
+        # assert len(self.produce_messages) == 3, f"Expected 3 messages, got {len(self.produce_messages)}"
+        # assert len(self.produce_errors) == 0, f"Expected 0 errors, got {len(self.produce_errors)}"
```

Comment:
THe idea is that these will be uncommented once paired with a working OIDC implementation in RP?

Reply:
Yep

Reply:
So my feeling with this is, uncommented them, mark the test `@ok_to_fail` and then remove it once OIDC is integrated.  Thoughts @BenPope ?

Reply:
Eh, maybe better to remove them? I still think the other assertions here have some value in terms of testing the test.

Reply:
Other option would be to just have this test be a simple service test (does this service actually work) and remove any comms with Redpanda and add them back later.

Reply:
Ripped a bunch of it out. Left a few things in to demonstrate service user manipulation and that librdkafka does the correct thing when passed the OAUTHBEARER algorithm, but the rest, I agree, was just noise.